### PR TITLE
Fixing error in train/val/test split contruction when using `graphstorm.gconstruct.construct_graph`

### PR DESCRIPTION
**Pull request by CongWeilin**
*Issue #, if available:*
Current implementation cannot handle multi-label classification's data split. For example, lets say we have 7 data points and 6 different labels, i.e., the label is a `7 * 6` numpy matrix (`np.float32`). We have 4 labeled data (as `0` or `1`) and 3 unlabeled data (as `np.NAN`):

```
label = 
array([[ 1.,  1.,  1.,  1.,  1.,  1.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 1.,  1.,  1.,  1.,  1.,  1.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan]], dtype=float32)
```

When using the original code `np.logical_not(np.isnan(label)).nonzero()[0]` ([code reference](https://github.com/awslabs/graphstorm/blob/c248f985cbe9b28cce8211cbe3446525d32d3a4d/python/graphstorm/gconstruct/transform.py#L327)) it will return 
```
array([0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3])
```

Then, this index will be passed to `self.data_split(permute_idx, len(label))` at [here](https://github.com/awslabs/graphstorm/blob/c248f985cbe9b28cce8211cbe3446525d32d3a4d/python/graphstorm/gconstruct/transform.py#L430), which will be later used to count how many labeled data at `num_labels = len(rand_idx)` ([code reference](https://github.com/awslabs/graphstorm/blob/c248f985cbe9b28cce8211cbe3446525d32d3a4d/python/graphstorm/gconstruct/transform.py#L382)). 

As a result, the code will assume we have `4 * 6 = 24` labeled data, but in fact it should only have `4` labeled data. 
 
*Description of changes:*

The code `np.nonzero(np.sum(np.isnan(label), axis=1) == 0)[0]`will only return `array([0, 1, 2, 3])` and ignore the duplicates.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/transform.py [hunk 1]
```diff
@@ -324,7 +324,10 @@ def get_valid_label_index(label):
     Numpy array : the index of the samples with valid labels in the list.
     """
     if np.issubdtype(label.dtype, np.floating):
-        return np.logical_not(np.isnan(label)).nonzero()[0]
+        if label.ndim == 1:
+            return np.logical_not(np.isnan(label)).nonzero()[0]
+        else:
+            return np.nonzero(np.sum(np.isnan(label), axis=1) < label.shape[1])[0]
```
[the remark this answers is not in the document]
Comment (CongWeilin) [reply python/graphstorm/gconstruct/transform.py:330]: I add https://github.com/awslabs/graphstorm/pull/210/commits/f3e7977dae1db1120e677c1b3a8aba107a0228dd in `tests/unit-tests/test_construct_graph.py`

File: tests/unit-tests/test_construct_graph.py [hunk 2]
```diff
@@ -300,6 +300,13 @@ def check_classification(res):
     res = process_labels(data, ops)
     check_classification(res)
 
+    # Check multi-label classification with invalid labels.
+    data = {'label' : np.random.randint(2, size=(13,5)).astype(np.float32)}
+    data['label'][[0, 3, 4]] = np.NAN
```
Comment (classicsong) [inline tests/unit-tests/test_construct_graph.py:305]: Can you also check the case when data['label'][0][0] = np.NAN but data['label'][0][1] is a valid number? I think your code also handles this case.

File: python/graphstorm/gconstruct/transform.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CongWeilin) [reply python/graphstorm/gconstruct/transform.py:330]: Please check https://github.com/awslabs/graphstorm/pull/210/commits/bcaaed8ff84692fa3963eea07ec1bf6268663894

File: tests/unit-tests/test_construct_graph.py (hunk 2 above)
Comment (CongWeilin) [reply tests/unit-tests/test_construct_graph.py]: Please check https://github.com/awslabs/graphstorm/pull/210/commits/bcaaed8ff84692fa3963eea07ec1bf6268663894

**Review by classicsong — APPROVED** [review]
LGTM

File: python/graphstorm/gconstruct/transform.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py:330]: It seems you should check whether np.sum(np.isnan(label), axis=1) == 0.

**Pull request merged**